hydrolib/tools/prof_to_cross.py

```diff
@@ -0,0 +1,545 @@
+    profdef.txt file into a CrossSectionDefinition (sub)class.
+
+    The resulting crosssection definition can be stored in the official
+    .ini file format.
+
+    Args:
+        profdef (dict): the key-value pairs read from profdef.txt. Must
+            contain several keys, depending on type, but at least:
+            PROFNR, TYPE, WIDTH, and optionally: ZMIN, BASE, TALUD,
+            FRCTP and FRCCF (all in uppercase).
+        profdefxyzfile (Optional[PolyFile]): A PolyFile (.pliz) containing
+            the x,y,z polyline points for all xyz profile definitions
+            (i.e., type=200/201). Polyline label must equal "PROFNR=<N>",
+            corresponding to the same labels in the profdef.txt file.
+
+    Returns:
+        CrossSectionDefinition: a cross section definition object, already
+            of the correct subclass, depending on the type, e.g., YZCrsDef.
+
+    Raises:
+        ValueError: when profdef has a wrong profile type, or when xyz data
+            is missing.
+    """
+
+    (proftype, crstype) = _proftype_to_crsdeftype(profdef.get("TYPE"))
+
+    crsdata = {}
+    if proftype == 1:
+        crsdata = _convert_pipe_definition(profdef)
+    elif proftype == 2:
+        crsdata = _convert_rectangle_definition(profdef)
+    elif proftype == 4:
+        crsdata = _convert_v_shape_definition(profdef)
+    elif proftype == 6:
+        crsdata = _convert_trapezoid_definition(profdef)
+    elif proftype == 100:
+        raise ValueError(
+            f"Type YZ found in input for PROFNR={profdef['PROFNR']}. Use XYZ instead."
+        )
+    elif proftype == 200:
+        if profdefxyzfile is None:
+            raise ValueError(
+                f"Profdefxyzfile is missing. Required for xyz profile type (PROFNR={profdef['PROFNR']})."
+            )
+
+        # Find the single polyline object in profdefxyz file for this PROFNR
+        polyobj = next(
+            (
+                polyobj
+                for polyobj in profdefxyzfile.objects
+                if polyobj.metadata.name == f"PROFNR={profdef['PROFNR']}"
+            )
+        )
+
+        crsdata = _convert_xyz_definition(profdef, polyobj)
+
+    _crs_append_friction_definition(
+        profdef, crsdata, plural=(crstype == "yz" or crstype == "xyz")
+    )
+
+    crsdata["id"] = f"PROFNR{profdef['PROFNR']}"
+    crsdata["type"] = crstype
+    if crstype == "circle":
+        crsdeftype = CircleCrsDef
+    elif crstype == "rectangle":
+        crsdeftype = RectangleCrsDef
+    elif crstype == "yz":
+        crsdeftype = YZCrsDef
+    elif crstype == "xyz":
+        crsdeftype = XYZCrsDef
+    else:
+        crsdeftype = CrossSectionDefinition
+
+    crsdef = crsdeftype(**crsdata)
+
+    return crsdef
+
+
+def prof_to_cross_from_mdu(
+    mdufile: PathOrStr,
+    crslocfile: PathOrStr = None,
+    crsdeffile: PathOrStr = None,
+):
+    """Wrapper converter function for converting legacy profile files
```

Comment:
Do we need to state that it is a wrapper, is that information usefull for the user?

Reply:
"Wrapper" can go for me, but it's to stress that this has MDU as input, but for the rest is the same as the actual converter function. Not that "user" is a "developer" here.